inc/sitemaps/class-sitemaps.php

```diff
@@ -0,0 +1,436 @@
+<?php
+/**
+ * @package WPSEO\XML_Sitemaps
+ */
+
+/**
+ * Class WPSEO_Sitemaps
+ */
+class WPSEO_Sitemaps {
+
+	/** @var string $sitemap Content of the sitemap to output. */
+	protected $sitemap = '';
+
+	/** @var string $stylesheet XSL stylesheet for styling a sitemap for web browsers. */
+	private $stylesheet = '';
+
+	/** @var bool $bad_sitemap Flag to indicate if this is an invalid or empty sitemap. */
+	public $bad_sitemap = false;
+
+	/** @var bool $transient Whether or not the XML sitemap was served from a transient or not. */
+	private $transient = false;
+
+	/** @var int $max_entries The maximum number of entries per sitemap page. */
+	private $max_entries;
+
+	/** @var string $http_protocol HTTP protocol to use in headers. */
+	protected $http_protocol = 'HTTP/1.1';
+
+	/** @var int $current_page Holds the n variable. */
+	private $current_page = 1;
+
+	/** @var WPSEO_Sitemap_Timezone $timezone */
+	private $timezone;
+
+	/** @var WPSEO_Sitemaps_Router $router */
+	public $router;
+
+	/** @var WPSEO_Sitemaps_Renderer $renderer  */
+	public $renderer;
+
+	/** @var WPSEO_Sitemap_Provider[] $providers */
+	public $providers;
+
+	/**
+	 * Class constructor
+	 */
+	public function __construct() {
+
+		if ( ! defined( 'ENT_XML1' ) ) {
+			define( 'ENT_XML1', 16 );
+		}
+
+		add_action( 'after_setup_theme', array( $this, 'reduce_query_load' ), 99 );
+		add_action( 'pre_get_posts', array( $this, 'redirect' ), 1 );
+		add_action( 'wpseo_hit_sitemap_index', array( $this, 'hit_sitemap_index' ) );
+
+		$stylesheet_url    = preg_replace( '/(^http[s]?:)/', '', esc_url( home_url( 'main-sitemap.xsl' ) ) );
+		$this->stylesheet  = '<?xml-stylesheet type="text/xsl" href="' . $stylesheet_url . '"?>';
```

Comment:
Hmm, I see now that it's not a default but an initial state indeed... But still, I dislike string building like this being done inside a constructor. However, after looking somewhat more into this, I've come to a different conclusion, which is that the stylesheet logic doesn't belong here in the first place. Please deprecate `set_stylesheet` and refactor it to use the `wpseo_stylesheet_url`. Then move the stylesheet default to the `WPSEO_Sitemaps_Renderer` and remove the `stylesheet` property from `WPSEO_Sitemaps` altogether.